### PR TITLE
Arnej/render empty strings

### PR DESCRIPTION
**Pull request by arnej27959**
@geirst or @baldersheim please review and merge
@bratseth FYI

### CODE REVIEW

File: vespabase/src/start-cbinaries.sh [hunk 1]
```diff
@@ -92,7 +92,7 @@ check_bname_in_value () {
 
 configure_valgrind () {
     no_valgrind=true
-    if which valgrind >/dev/null; then
+    if which valgrind >/dev/null 2>&1; then
         if check_bname_in_value $VESPA_USE_VALGRIND; then
```
Comment (baldersheim) [inline vespabase/src/start-cbinaries.sh]: Intended ?

File: document/src/main/java/com/yahoo/document/json/JsonSerializationHelper.java [hunk 2]
```diff
@@ -116,11 +116,10 @@ public static void serializeReferenceField(JsonGenerator generator, FieldBase fi
     }
 
     public static void serializeStringField(JsonGenerator generator, FieldBase field, StringFieldValue value) {
-        // Hide empty strings
-        if (value.getString().length() == 0) {
+        // Hide fields which only contains an empty string
+        if (value.getString().length() == 0 && field != null) {
             return;
```
Comment (baldersheim) [inline document/src/main/java/com/yahoo/document/json/JsonSerializationHelper.java]: I think I need some context on this one.

Comment (baldersheim) [reply document/src/main/java/com/yahoo/document/json/JsonSerializationHelper.java]: When serializing an entry in an array 'field' will be null. Hence giving correct behavior for arrays.

File: vespabase/src/start-cbinaries.sh (hunk 1 above)
Comment (baldersheim) [reply vespabase/src/start-cbinaries.sh]: I see it was a separate commit so then the answer is yes.

**Review by baldersheim — APPROVED** [review]
I see no changes to any tests. I guess this could be tested easily in a unit test.

**Review by geirst — COMMENTED** [review]
I would like to see a unit test as well. Otherwise LGTM

**Comment by arnej27959** [comment]
@baldersheim please take another look

**Review by baldersheim — APPROVED** [review]

File: document/src/main/java/com/yahoo/document/json/JsonSerializationHelper.java (hunk 2 above)
Comment (bratseth) [reply document/src/main/java/com/yahoo/document/json/JsonSerializationHelper.java]: That context should be in the code ...

**Pull request merged**